laws/applicative.js

```diff
@@ -1,36 +1,26 @@
 'use strict';
 
-const {identity, thrush} = require('fantasy-combinators');
 const {of, ap} = require('..');
 
-/**
-
-### Applicative
-
-1. `v.ap(A.of(x => x))` is equivalent to `v` (identity)
-2. `A.of(x).ap(A.of(f))` is equivalent to `A.of(f(x))` (homomorphism)
-3. `A.of(y).ap(u)` is equivalent to `u.ap(A.of(f => f(y)))` (interchange)
-
-**/
-
-const identityʹ = T => eq => x => {
-  const a = T[of](x)[ap](T[of](identity));
-  const b = T[of](x);
-  return eq(a, b);
-};
-
-const homomorphism = T => eq => x => {
-  const a = T[of](x)[ap](T[of](identity));
-  const b = T[of](identity(x));
-  return eq(a, b);
-};
-
-const interchange = T => eq => x => {
-  const u = T[of](identity);
-
-  const a = T[of](x)[ap](u);
-  const b = u[ap](T[of](thrush(x)));
-  return eq(a, b);
-};
-
-module.exports = {identity: identityʹ, homomorphism, interchange};
+const identity = eq => (A, v) => {
+  return eq(
+    v[ap](A[of](x => x)),
+    v
+  );
+}
+
+const homomorphism = eq => (A, x, f) => {
```

Comment:
What about moving that up further, to the exports?

```js
// laws/monoid.js
const rightIdentity = (X, eq) => m => ...

module.exports = (X, eq) => {
  return {
    rightIdentity: rightIdentity(X, eq),
  };
};
```

```js
// laws/traversable.js
const composition = (X, eq) => (F, G, u) => ...

module.exports = (X, eq) => {
  return {
    composition: composition(X, eq),
  };
};
```

```js
// laws.js
const Monoid = require ('./laws/monoid');
const Traversable = require ('./laws/traversable');

module.exports = (X, eq) => {
  return {
    Monoid: Monoid(X, eq),
    Traversable: Traversable(X, eq),
  };
};
```

Or however it's supposed to be structured. Users can just import the top level laws thing, and unpack from there. That gives the added benefit that all the things are the same across laws. Or they can pull in piece meal if they need that for some reason–which is how it currently is.